pkg/ddevapp/router.go

```diff
@@ -349,10 +350,41 @@ func determineRouterPorts() []string {
 		util.Failed("Failed to retrieve containers for determining port mappings: %v", err)
 	}
 
-	// Add potential ports that might not be in use by current containers
-	// 8142/8143 are for xhgui, which might not be enabled, but we don't want to
-	// have to rebuild traefik when it does get enabled.
-	routerPorts = []string{nodeps.DdevDefaultXHGuiHTTPSPort, nodeps.DdevDefaultXHGuiHTTPPort}
```

Comment:
This PR makes port detection dynamic, so we no longer need to hardcode XHGui.

Though if the service isn't started, it's unclear why the ports should be occupied.

But this comment gives some justification:

> we don't want to have to rebuild traefik when it does get enabled.

I didn't look deep into it.